### PR TITLE
[uss_qualifier/geo-awareness] Update and generalize geo-awareness test API

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
This PR makes some changes to the geo-awareness test API to:

1. Make it more broadly applicable to geo-awareness in non-U-space/ED-269 contexts
2. Clarify/unify some concepts (especially "selection of Geozones based on filter criteria")
3. Harmonize terminology between "UAS Zone" and "Geozone" (selecting "Geozone" as the more generic term)
4. Remove `/v1` prefixes
5. Make a few other smaller adjustments.

### CODE REVIEW

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 1]
```diff
@@ -147,57 +147,100 @@ components:
...
       properties:
         uomDimensions:
           $ref: '#/components/schemas/UomDimensions'
         verticalReferenceType:
           $ref: '#/components/schemas/VerticalReferenceType'
         verticalReference:
+          description: >-
+            Height above vertical reference datum, in units of uomDimensions.
           type: number
+          default: 0
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:159]: Some serialization protocols can't differentiate between omission and the default value of a field (Protobuf).  IMO, this was a very questionable design decision, but they are in use so it's generally best to accommodate their limitations when practical.  Best practice for this is generally to make otherwise-required fields optional with a default value that matches the standard default value for that data type.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 2]
```diff
@@ -147,57 +147,100 @@ components:
...
         - verticalReferenceType
-        - verticalReference
-        - longitude
-        - latitude
       properties:
         uomDimensions:
           $ref: '#/components/schemas/UomDimensions'
         verticalReferenceType:
           $ref: '#/components/schemas/VerticalReferenceType'
         verticalReference:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:155]: Is the field name `verticalReference` used in ED-269?  If not, this should probably be something like `heightAboveVerticalReference` rather than simply `verticalReference` (since the reference datum is actually named in `verticalReferenceType` rather than this field)

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 3]
```diff
@@ -147,57 +147,100 @@ components:
...
           type: number
+          default: 0
         latitude:
+          description: >-
+            Latitude, degrees north of the equator.
           type: number
+          default: 0
 
-    UASZonesCheckRequest:
+    GeozonesCheckRequest:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:171]: I've changed all instances of "UAS Zone" to "Geozone" as I think "Geozone" is more generic (ED-269 UASZones are special cases of generic Geozones).

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 4]
```diff
@@ -147,57 +147,100 @@ components:
...
       type: object
       required:
-        - position
-        - uSpaceClass
-        - acceptableRestrictions
-        - startDateTime
-        - endDateTime
+        - checks
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:174]: Since I think we'll have a number of tests where we'll want to perform a number of different checks, this saves a bunch of HTTP handshaking to perform multiple checks in a single HTTP call.  The response now has a corresponding array of results.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 5]
```diff
@@ -147,57 +147,100 @@ components:
...
-            $ref: '#/components/schemas/Restriction'
-        startDateTime:
+            $ref: '#/components/schemas/GeozonesCheck'
+
+    GeozonesCheck:
+      type: object
+      required:
+        - filterSets
+      properties:
+        filterSets:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:186]: This field name is intended to make it clear that the USSP should be filtering its geo-awareness data according to the criteria provided in order to make the final true/false determination.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 6]
```diff
@@ -147,57 +147,100 @@ components:
...
+      description: >-
+        Set of filters to select only a subset of Geozones.  Only Geozones which are applicable to all specified filters within this filter set should be selected.
+      type: object
+      properties:
+        position:
+          description: >-
+            If specified, only select Geozones encompassing this position.
+          anyOf:
+            - $ref: '#/components/schemas/Position'
+        after:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:203]: I think this is clearer than `startDateTime` because nothing starts at this time -- we are interested in some Geozones that start before this time, and some Geozones that start after this time.  The important thing is that we only look for Geozones that some temporal part that extends past this time.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 7]
```diff
@@ -147,57 +147,100 @@ components:
...
+            If specified, only select Geozones which are defined at or after this time.
           type: string
           format: date-time
-        endDateTime:
+        before:
+          description: >-
+            If specified, only select Geozones which are defined at or before this time.
           type: string
           format: date-time
+        ed269:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:213]: By grouping filter criteria by requirement source, we can add filter criteria for other contexts alongside existing filter criteria without affecting existing filter criteria.  When the replacement for ED-269 is release, we can leave this field and the tests built for it while still stepping up to and supporting the new ED-xxx by simply adding a new filter criteria group like `edXXX`.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 8]
```diff
@@ -147,57 +147,100 @@ components:
...
           description: >-
-            True if there is one or more applicable UASZone under the requested conditions.
-        error:
-          description: Human-readable explanation of the result for debugging purpose only. This field is required when the check do not complete.
+            True if and only if one or more applicable Geozones were selected according to the specified selection criteria.
+          default: false
 
 paths:
-  /v1/status:
+  /status:
```
Comment (BenjaminPelletier) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:247]: Although other APIs include a version prefix like this, I think we shouldn't do that going forward.  USSPs are free to make the base URL whatever they want, so mandating a shared prefix like this is unnecessary and overly-constraining.  I suspect a common desire would be to include the API version in the base URL, and that API version is a full semantic version (like 0.1.0) rather than simply "v1" (there are multiple "v1"s).

**Review by BenjaminPelletier — COMMENTED** [review]
I've added some contextual comments to help with review.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml (hunk 2 above)
Comment (barroco) [reply interfaces/automated-testing/geo-awareness/geo-awareness.yaml:155]: No indeed you are right, it uses the following fields:
- lowerLimit
- lowerVerticalReference
- upperLimit
- upperVerticalReference

`verticalReference` with type `VerticalReferenceType`
and
`heighAboveVerticalReference` with type number would be correct.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml (hunk 8 above)
Comment (barroco) [reply interfaces/automated-testing/geo-awareness/geo-awareness.yaml:247]: Sounds good to me.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml (hunk 7 above)
Comment (barroco) [reply interfaces/automated-testing/geo-awareness/geo-awareness.yaml:213]: This is a great improvement to the initial design. Thank you!

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml [hunk 9]
```diff
@@ -147,57 +147,104 @@ components:
       required:
         - uomDimensions
         - verticalReferenceType
-        - verticalReference
-        - longitude
-        - latitude
```
Comment (barroco) [inline interfaces/automated-testing/geo-awareness/geo-awareness.yaml:152]: Why did you remove the requirement on those fields?

**Review by barroco — APPROVED** [review]
Thank you for the improvements and detailed comments. Please check my answers and question below.

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml (hunk 9 above)
Comment (BenjaminPelletier) [reply interfaces/automated-testing/geo-awareness/geo-awareness.yaml:152]: [To accommodate serializers that can't differentiate between omission and the default value](https://github.com/interuss/dss/pull/818#discussion_r993964212)

File: interfaces/automated-testing/geo-awareness/geo-awareness.yaml (hunk 2 above)
Comment (BenjaminPelletier) [reply interfaces/automated-testing/geo-awareness/geo-awareness.yaml]: I've now updated the `verticalReference` field name.
